src/main/java/org/terasology/launcher/config/GameConfig.java

```diff
@@ -0,0 +1,172 @@
+        return dataDir;
+    }
+
+    public JavaHeapSize getMaxMemory() {
+        return maxMemory;
+    }
+
+    public JavaHeapSize getInitMemory() {
+        return initMemory;
+    }
+
+    public String getJavaParam() {
+        return javaParam;
+    }
+
+    public String getGameParam() {
+        return gameParam;
+    }
+
+    public LogLevel getLogLevel() {
+        return logLevel;
+    }
+
+    /**
+     * Provides a pre-filled {@link Builder} instance
+     * with all configurations copied from this. Use it
+     * to create a new {@link GameConfig} instance with
+     * modified configurations.
+     *
+     * @return a pre-filled {@link Builder} instance
+     */
+    public Builder rebuilder() {
+        return new Builder(this);
+    }
+
+    /**
+     * Provides an empty {@link Builder} instance to fill
+     * configurations using builder pattern.
+     *
+     * @return an empty {@link Builder} instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Nested builder class to create {@link GameConfig} instances
+     * using descriptive methods.
+     */
+    public static final class Builder {
+        private Path installDir;
+        private Path dataDir;
+        private JavaHeapSize maxMemory;
+        private JavaHeapSize initMemory;
+        private String javaParam;
+        private String gameParam;
+        private LogLevel logLevel;
+
+        private Builder() {
+            javaParam = "";
+            gameParam = "";
+        }
+
+        private Builder(final GameConfig last) {
+            installDir = last.installDir;
+            dataDir = last.dataDir;
+            maxMemory = last.maxMemory;
+            initMemory = last.initMemory;
+            javaParam = last.javaParam;
+            gameParam = last.gameParam;
+            logLevel = last.logLevel;
+        }
+
+        public Builder installDir(final Path newInstallDir) {
+            installDir = newInstallDir;
+            return this;
+        }
+
+        public Builder dataDir(final Path newDataDir) {
+            dataDir = newDataDir;
+            return this;
+        }
+
+        public Builder maxMemory(final JavaHeapSize newMaxMemory) {
+            maxMemory = newMaxMemory;
+            return this;
+        }
+
+        public Builder initMemory(final JavaHeapSize newInitMemory) {
+            initMemory = newInitMemory;
+            return this;
+        }
+
+        public Builder javaParam(final String newJavaParam) {
+            javaParam = (newJavaParam != null) ? newJavaParam : "";
+            return this;
+        }
+
+        public Builder gameParam(final String newGameParam) {
+            gameParam = (newGameParam != null) ? newGameParam : "";
+            return this;
+        }
+
+        public Builder logLevel(final LogLevel newLogLevel) {
+            logLevel = newLogLevel;
+            return this;
+        }
+
+        public GameConfig build() {
```

Comment:
Here, more parameters are mandatory, which makes the builder not as easy to use ... maybe we can think about default values? Alternatively, there are a couple of interesting ideas in [this SO thread](https://stackoverflow.com/questions/7302891/the-builder-pattern-and-a-large-number-of-mandatory-parameters).